From bdb3b4fbfb696e98cfb09077d1fa443b1f314eb3 Mon Sep 17 00:00:00 2001
From: William Wu <william.wu@rock-chips.com>
Date: Mon, 8 Apr 2019 14:39:10 +0800
Subject: [PATCH] dt-bindings: usb-xhci: add xhci-slow-suspend property

Add a new property "xhci-slow-suspend" for some xHCI controllers
(e.g. Rockchip SoCs) which are integrated in DWC3 IP, they need
an extraordinary delay to wait for xHCI enter the Halted state
(i.e. HCH in the USBSTS register is '1'), especially if DWC3 is
in DRD mode.

Change-Id: Iffda2125ce403dd2ea9ca47b779580bdc3303b41
Signed-off-by: William Wu <william.wu@rock-chips.com>
---
 Documentation/devicetree/bindings/usb/usb-xhci.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Documentation/devicetree/bindings/usb/usb-xhci.txt b/Documentation/devicetree/bindings/usb/usb-xhci.txt
index ac4cd0d6195a..dab51612eb8f 100644
--- a/Documentation/devicetree/bindings/usb/usb-xhci.txt
+++ b/Documentation/devicetree/bindings/usb/usb-xhci.txt
@@ -36,6 +36,8 @@ Optional properties:
   - usb2-lpm-disable: indicate if we don't want to enable USB2 HW LPM
   - usb3-lpm-capable: determines if platform is USB3 LPM capable
   - quirk-broken-port-ped: set if the controller has broken port disable mechanism
+  - xhci-slow-suspend: set if it needs an extraordinary delay to wait
+    for xHC enter the Halted state after the Run/Stop (R/S) bit is cleared to '0'
   - imod-interval-ns: default interrupt moderation interval is 5000ns
   - phys : see usb-hcd.txt in the current directory
 
-- 
2.35.3

